include/zephyr/bluetooth/services/ias.h

```diff
@@ -60,6 +68,60 @@ int bt_ias_local_alert_stop(void);
 #define BT_IAS_CB_DEFINE(_name)                                                                    \
 	static const STRUCT_SECTION_ITERABLE(bt_ias_cb, _CONCAT(bt_ias_cb_, _name))
 
+struct bt_ias_client_cb {
+	/** @brief Callback function for bt_ias_discover.
+	 *
+	 *  This callback is called when discovery procedure is complete.
+	 *
+	 *  @param conn Bluetooth connection object.
+	 *  @param err 0 on success, ATT error or negative errno otherwise
+	 */
+	void (*discover)(struct bt_conn *conn, int err);
+};
+
+/** @brief Set high alert level
+ *
+ *  @param conn Bluetooth connection object
+ *
+ *  @return Zero in case of success and error code in case of error.
+ *
+ */
+int bt_ias_client_alert_high(struct bt_conn *conn);
+
+/** @brief Set mild alert level
+ *
+ *  @param conn Bluetooth connection object
+ *
+ *  @return Zero in case of success and error code in case of error.
+ *
+ */
+int bt_ias_client_alert_mild(struct bt_conn *conn);
+
+/** @brief Set stop alert level
+ *
+ *  @param conn Bluetooth connection object
+ *
+ *  @return Zero in case of success and error code in case of error.
+ *
+ */
+int bt_ias_client_alert_stop(struct bt_conn *conn);
+
+/** @brief Discover Immediate Alert Service
+ *
+ *  @param conn Bluetooth connection object
+ *
+ *  @return Zero in case of success and error code in case of error.
+ */
+int bt_ias_discover(struct bt_conn *conn);
+
+/** @brief Register Immediate Alert Client callbacks
+ *
+ *  @param cb The callback structure
+ *
+ *  @return Zero in case of success and error code in case of error.
+ */
+int bt_ias_client_cb_register(const struct bt_ias_client_cb *cb);
```

Comment:
Not that there's anything wrong with this, but how come the server callbacks are statically registered with `BT_IAS_CB_DEFINE`, and the client callbacks are dynamically registered with `bt_ias_client_cb_register`?

Reply:
I think the main reason for doing this way on the client-side was actually looking at other clients' code and seeing that it was generally using dynamically registered callbacks. Do you think this will be more proper to do this in convention between IAS client and server? If so, would it be better to do generally dynamic or static callbacks?

Reply:
It's a good question. We generally haven't done much of statically defined callbacks in the host (that was only quite recently introduced at all), so all existing profiles simply use(d) the dynamical register. 

Going forward i am not sure what approach we'll take: The dynamic approach has the advantage of being able (at least if implemented) to de-register callbacks when it is no longer interested in it, which can be a small run-time performance optimization (as fewer callbacks are called). 

I am not sure what general approach we'll use going forward:
1) Statically registered callbacks
2) Dynamically registered callbacks
3) Both (i.e. like the `struct bt_conn_cb`)

This service is marked as experimental, so we can just go ahead with the dynamic approach now, but perhaps we'll need to agree on a general approach at some point (soon).

@jhedberg @alwa-nordic @asbjornsabo Any input?